readthedocs/config/config.py

```diff
@@ -828,9 +827,14 @@ def validate_python(self):
 
         python = {}
         with self.catch_validation_error('python.version'):
-            version = self.pop_config('python.version', 3)
+            version = self.pop_config('python.version', '3')
+            if version == 3.1:
+                # Special case for ``python.version: 3.10``,
+                # yaml will transform this to the numeric value of `3.1`.
+                # Save some frustration to users.
```

Comment:
I think this is the right call. I don't love it, but I can't think of any other way to not make this UX workable.